File: src/com/fenghuo/notes/ThingsFragment.java
package com.fenghuo.notes;

import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.view.ViewGroup.LayoutParams;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.AdapterView.OnItemLongClickListener;
import android.widget.Button;
import android.widget.GridView;
import android.widget.PopupWindow;

import com.fenghuo.notes.adapter.NoteAdapter;
import com.fenghuo.notes.bean.Note;
import com.fenghuo.notes.db.DBNoteHelper;

import java.util.List;


public class ThingsFragment extends FragmentExt implements OnClickListener,
        OnItemClickListener, OnItemLongClickListener {

    private Button btn_complete;// 结束编辑
    private GridView gv_list;// 列表
    private DBNoteHelper noteHelper;// 数据库
    private List<Note> list;// 数据集合
    private NoteAdapter adapter;// 适配器
    private PopupWindow pop_button;
    private View mPopBtnView;// 弹出
    // 按钮
    private Note mCurrentNote;// 当前的note
    private View mRootView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        mRootView = inflater.inflate(R.layout.fragment_things, null, false);
        gv_list = (GridView) mRootView.findViewById(R.id.gv_list);
        mRootView.findViewById(R.id.add_new_note).setOnClickListener(this);
        noteHelper = new DBNoteHelper(getActivity());
        list = noteHelper.Getlist();
        adapter = new NoteAdapter(getActivity(), list);
        gv_list.setAdapter(adapter);

        gv_list.setOnItemClickListener(this);
        gv_list.setOnItemLongClickListener(this);

        return mRootView;
    }

    @Override
    public void onDestroy() {
        noteHelper.Desdroy();
        super.onDestroy();
    }

    @Override
    public void onResume() {
        list.clear();
        list = noteHelper.Getlist();
        adapter.Update(list);
        super.onResume();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (adapter.isEdit()) {
            adapter.quitEdit();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }

    @Override
    public void onClick(View arg0) {
        switch (arg0.getId()) {
            // 跳转界面
            case R.id.add_new_note:
                Intent intent = new Intent(getActivity(), AddNoteActivity.class);
                startActivity(intent);
                break;
            // 结束
            case R.id.btn_compelete:
                btn_complete.setVisibility(View.GONE);
                adapter.quitEdit();
                pop_button.dismiss();
                break;
            default:
                break;
        }

    }

    @Override
    public boolean onItemLongClick(AdapterView<?> arg0, View arg1, int arg2,
                                   long arg3) {
        adapter.startEdit();
        showPopButton();
        return true;
    }

    @Override
    public void onItemClick(AdapterView<?> arg0, View arg1, int position,
                            long arg3) {
        mCurrentNote = list.get(position);
        Intent intent_edit = new Intent(getActivity(), EditNoteActivity.class);
        intent_edit.putExtra("noteid", mCurrentNote.getId());
        startActivity(intent_edit);
    }

    private void showPopButton() {

        mPopBtnView = getActivity().getLayoutInflater().inflate(
                R.layout.pop_button, null);
        btn_complete = (Button) mPopBtnView.findViewById(R.id.btn_compelete);
        btn_complete.setOnClickListener(this);
        pop_button = new PopupWindow(mPopBtnView, LayoutParams.WRAP_CONTENT,
                LayoutParams.WRAP_CONTENT);
        pop_button.showAtLocation(getActivity().findViewById(R.id.vp_content), Gravity.CENTER_HORIZONTAL | Gravity.BOTTOM, 0, 0);
        pop_button.update();
    }

}

File: src/com/fenghuo/notes/ThingsFragment.kt
package com.fenghuo.notes

import android.content.Intent
import android.os.Bundle
import android.view.KeyEvent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.AdapterView.OnItemLongClickListener
import android.widget.GridView
import com.fenghuo.notes.adapter.NoteAdapter
import com.fenghuo.notes.bean.Note
import com.mine.view.button.FloatingActionButton


class ThingsFragment : FragmentExt(), OnItemClickListener, OnItemLongClickListener {

    private var list: GridView? = null// 列表
    private var adapter: NoteAdapter? = null// 适配器
    private var mRootView: View? = null
    private var saveBtn: FloatingActionButton? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mRootView = inflater.inflate(R.layout.fragment_things, null, false)
        list = mRootView!!.findViewById<GridView>(R.id.gv_list) as GridView

        mRootView?.findViewById<View>(R.id.add_new_note)?.setOnClickListener {
            val intent = Intent(activity, AddNoteActivity::class.java)
            startActivity(intent)
        }

        saveBtn = mRootView?.findViewById(R.id.save)
        saveBtn?.setOnClickListener { quiteEdit() }

        adapter = NoteAdapter(activity)

        list?.adapter = adapter

        list?.onItemClickListener = this
        list?.onItemLongClickListener = this

        return mRootView
    }

    override fun onResume() {
        super.onResume()
        adapter?.Update();
    }

    override fun onDestroy() {
        super.onDestroy()
        adapter?.destory()
    }

    private fun quiteEdit() {
        adapter!!.quitEdit()
        saveBtn!!.hide(true)
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (adapter!!.isEdit) {
            quiteEdit()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

    override fun onItemLongClick(arg0: AdapterView<*>, arg1: View, arg2: Int, arg3: Long): Boolean {
        adapter!!.startEdit()
        saveBtn!!.show(true)
        return true
    }

    override fun onItemClick(arg0: AdapterView<*>, view: View, position: Int,
                             arg3: Long) {
        val note: Note? = view.getTag(R.id.tag_1) as Note
        note?.let {
            val intent = Intent(activity, EditNoteActivity::class.java)
            intent.putExtra("noteid", it.id)
            startActivity(intent)
        }
    }

}
